File: app/src/main/java/com/androidexamples/app/activity/GesturesActivity.java
package com.androidexamples.app.activity;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;

import com.androidexamples.app.R;

public class GesturesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_gestures);
        LoadComponents();
    }

    private void LoadComponents() {
        setSupportActionBar((Toolbar)findViewById(R.id.top_toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        findViewById(R.id.button_zoomable_imageview).setOnClickListener(onZoomableImageViewButtonClick());
        findViewById(R.id.button_drag_drop_image).setOnClickListener(onDragDropImageButtonClick());
    }

    private View.OnClickListener onDragDropImageButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getBaseContext(), DraggableViewActivity.class));
            }
        };
    }

    private View.OnClickListener onZoomableImageViewButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getBaseContext(), ZoomableImageActivity.class));
            }
        };
    }
}

File: app/src/main/java/com/androidexamples/app/flows/gestures/GesturesActivity.kt
package com.androidexamples.app.flows.gestures

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.view.View

import com.androidexamples.app.R

class GesturesActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_gestures)
        LoadComponents()
    }

    private fun LoadComponents() {
        setSupportActionBar(findViewById<View>(R.id.top_toolbar) as Toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        findViewById<View>(R.id.button_zoomable_imageview).setOnClickListener(onZoomableImageViewButtonClick())
        findViewById<View>(R.id.button_drag_drop_image).setOnClickListener(onDragDropImageButtonClick())
    }

    private fun onDragDropImageButtonClick(): View.OnClickListener {
        return View.OnClickListener { startActivity(Intent(baseContext, DraggableViewActivity::class.java)) }
    }

    private fun onZoomableImageViewButtonClick(): View.OnClickListener {
        return View.OnClickListener { startActivity(Intent(baseContext, ZoomableImageActivity::class.java)) }
    }
}
